### PR TITLE
support new meas_date of type datetime in mne master

### PR DESCRIPTION
**Pull request by agramfort**
deal https://github.com/mne-tools/mne-python/pull/7124

@sappelhoff @jasmainak I have a pb though as I don't know how to deal with annotations whose duration is `'n/a`.

what is the semantic of this?

### CODE REVIEW

**Comment by jasmainak** [comment]
humm ... I suspect you should skip those events.

see https://github.com/mne-tools/mne-bids/pull/234/. I don't fully understand the PR now. Because when you write events, the duration should be 0. And when reading, if it's not 0, then we can't handle it in MNE, so we should skip it. Agree @sappelhoff ?

File: mne_bids/read.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def _handle_events_reading(events_fname, raw):
 
     # Deal with "n/a" strings before converting to float
     ons = [np.nan if on == 'n/a' else on for on in events_dict['onset']]
-    dus = [np.nan if du == 'n/a' else du for du in events_dict['duration']]
+    dus = [0. if du == 'n/a' else du for du in events_dict['duration']]
```
Comment (agramfort) [inline mne_bids/read.py]: @sappelhoff this is what I had to change. there is no concept of NaN for datetime. Why supporting n/a durations ? is it safe to make it 0 ?

**Comment by sappelhoff** [comment]
Mmh it's difficult ... in BIDS, durations of length `0` are handled as "instantaneous events" ... coming from the MRI background, so that these events can be "modeled with a single impulse function":

> A "duration" value of zero implies that the delta function or event is so short as to be effectively modeled as an impulse.

(see: https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/05-task-events.html)

so turning `np.nan` to `0` is probably not safe.

I guess skipping and sending a warning might be best. So users can see why the events are skipped ... and then the user can evaluate whether it makes sense to turn these durations to `0` or not.

**Comment by agramfort** [comment]
ok I'll revert the nan change once https://github.com/mne-tools/mne-python/pull/7131/files is in.

**Comment by jasmainak** [comment]
LGTM, thanks @agramfort !

**Pull request merged**